src/toxcore/messenger/packet/mod.rs

```diff
@@ -73,14 +78,14 @@ impl FromBytes for Packet {
         map!(Online::from_bytes, Packet::Online) |
         map!(Action::from_bytes, Packet::Action) |
         map!(Offline::from_bytes, Packet::Offline) |
-        map!(Message::from_bytes, Packet::Message) |
         map!(Nickname::from_bytes, Packet::Nickname) |
+        map!(Message::from_bytes, Packet::Message) |
         map!(UserStatus::from_bytes, Packet::UserStatus) |
         map!(FileControl::from_bytes, Packet::FileControl) |
         map!(FileData::from_bytes, Packet::FileData) |
+        map!(Msi::from_bytes, Packet::Msi) |
         map!(StatusMessage::from_bytes, Packet::StatusMessage) |
-        map!(Typing::from_bytes, Packet::Typing) |
-        map!(FileData::from_bytes, Packet::FileData)
```

Comment:
FileData?

Reply:
https://github.com/tox-rs/tox-spec/blob/master/spec.md#file_data

Reply:
You removed it from parsing. Should you keep it?

Reply:
Sorry. There was many PRs on this file. So the order of map! is scrambled.
FileData is here.
https://github.com/tox-rs/tox/blob/ed9250295a254b30af1682f13f214c4f002d205c/src/toxcore/messenger/packet/mod.rs#L85

Reply:
OK